packages/cli/src/vercel-snapshot.ts

```
@@ -0,0 +1,53 @@
+import path from 'path';
+import { globby } from 'globby';
+import fsExtra from 'fs-extra';
+
+/**
+ * When deploying a mosaic site to vercel, snapshot files are ignored resulting in a 404 when a page is accessed from the site.
+ *
+ * Vercel uses output file tracing (https://nextjs.org/docs/advanced-features/output-file-tracing) to determine what files need to be deployed.
+ * Due to the dynamic nature of mosaic sites, files that are required, the snapshots, are not traced correctly.
+ *
+ * This script adds an entry into the [...route].js.nft.json file for each file in the provided snapshot ensuring that the snapshot files are deployed.
+ *
+ */
+export async function updateTraceFile(config, options) {
+  const { mode, platform } = config.deployment;
+
+  if (mode === 'snapshot-file' && platform === 'vercel') {
```

should we move this check upto index.ts or add code to prevent a silent fail